ui/app/routes/topology.js

```diff
@@ -28,4 +28,17 @@ export default class TopologyRoute extends Route.extend(WithForbiddenState) {
       nodes: this.store.query('node', { resources: true }),
     }).catch(notifyForbidden(this));
   }
+
+  setupController(controller, model) {
+    // When the model throws, make sure the interface expected by the controller is consistent.
+    if (!model) {
+      controller.model = {
+        jobs: [],
+        allocations: [],
+        nodes: [],
+      };
+    }
+
+    return super.setupController(...arguments);
+  }
```

Comment:
I don’t know how this would apply here but I’m reminded of [this tweet](https://twitter.com/mixonic/status/1324357014621925377) I saw this morning

Reply:
I totally agree with that tweet, and the same thought crossed my mind here. I didn't end up doing anything about it because I didn't want to accidentally over-engineer anything, but this is definitely a pattern we use elsewhere.